stbt_auto_selftest.py

```diff
@@ -92,23 +96,33 @@ def main(argv):
     os.chdir(root)
 
     if cmdline_args.command == 'generate':
-        generate()
+        generate(cmdline_args.source_files)
     elif cmdline_args.command == 'validate':
         return validate()
     else:
         assert False
 
 
-def generate():
-    tmpdir = generate_into_tmpdir()
+def generate(source_files):
+    valid_source_files = _valid_source_files(source_files)
```

Comment:
You probably want to error for any source files that don't exist.  You may wish to error for any source files that won't or don't produce an auto-selftest file.  This would probably be best for interactive use.